app/controllers/api/v1/profile_providers_controller.rb

```diff
@@ -14,8 +14,7 @@ def create
           render json: { message: 'Provider already linked' }, status: :internal_server_error
           return
         end
-        state = { provider_id: provider.id, profile_id: profile.id }.to_json
-        state_enc = Base64.encode64(state)
+        state_enc = "#{provider.id}:#{profile.id}"
```

Comment:
Could you move the encode/decode impl to the HDM::OAuth::State class? Or alternatively just remove that class to avoid confusion. I think the failing test case is referencing that class.